scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -361,19 +361,32 @@ def func(x):
         assert_allclose(ret_bounds_list.fun, ret_bounds_class.fun, atol=1e-9)
         assert ret_bounds_list.nfev == ret_bounds_class.nfev
 
-    def test_callable_jac_with_args_gh11052(self):
+    def test_callable_jac_hess_with_args_gh11052(self):
         # dual_annealing used to fail when `jac` was callable and `args` were
         # used; check that this is resolved. Example is from gh-11052.
+
+        # extended to hess as part of closing gh20614
         rng = np.random.default_rng(94253637693657847462)
         def f(x, power):
             return np.sum(np.exp(x ** power))
 
         def jac(x, power):
             return np.exp(x ** power) * power * x ** (power - 1)
 
+        def hess(x, power):
+            # calculated using WolframAlpha as d^2/dx^2 e^(x^p)
```

Comment:
Thanks!
<img width="687" alt="image" src="https://github.com/scipy/scipy/assets/6570539/3e2522d5-dfb6-4aba-9e3c-561fa32d1b9e">
👍

Reply:
I did it by hand first :-). Do you use WolframAlpha at all?

Reply:
All the time to get reference values of special/stats functions. Sometimes need to use Mathematics proper if it's not provided with Alpha.